ambry-rest/src/main/java/com.github.ambry.rest/NettyMessageProcessor.java

```diff
@@ -138,20 +138,22 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
       throws Exception {
     try {
       nettyMetrics.processorExceptionCaughtCount.inc();
-      if (cause instanceof RestServiceException) {
-        RestServiceErrorCode errorCode = ((RestServiceException) cause).getErrorCode();
-        if (ResponseStatus.getResponseStatus(errorCode) == ResponseStatus.BadRequest) {
-          logger.debug("Error on channel {}", ctx.channel(), errorCode, cause);
+      if (request != null && request.isOpen() && cause instanceof Exception) {
+        responseChannel.onResponseComplete((Exception) cause);
+      } else {
+        if (cause instanceof RestServiceException) {
```

Comment:
can you help me understand this. I don't get it. "Else" will be executed for those that are throwable but not an instance of Exception. But RestServiceException is a subclass of Exception right. Am I missing something here?